lib/elixir/lib/code/typespec.ex

```diff
@@ -273,13 +273,16 @@ defmodule Code.Typespec do
           {{:optional, [], [typespec_to_quoted(k)]}, typespec_to_quoted(v)}
       end)
 
-    {struct, fields} = Keyword.pop(fields, :__struct__)
-    map = {:%{}, [line: line], fields}
+    with {:__struct__, struct} <- List.keyfind(fields, :__struct__, 0),
```

Comment:
You can also use case+List.keytake. I think that will be cleaner:

```suggestion
    case List.keytake(fields, :__struct__, 0) do
      {{:__struct__, struct}, fields_pruned} when is_atom(struct) and struct != nil ->
        {:%, [line: line], [struct, fields_pruned]}

      _ ->
        {:%{}, [line: line], fields}
    end
```